aiven/internal/mutexkv/mutexkv.go

```diff
@@ -0,0 +1,51 @@
+package mutexkv
```

Comment:
we already gave a `pkg` package, perhaps it makes sense to move this file over there

Reply:
Also, I was thinking that this whole concurrency thing can be simplified with something like this:
```go
var esACLSem = semaphore.NewWeighted(1)

func resourceElasticsearchACLUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
	if !esACLSem.TryAcquire(1) {
		log.Printf("[TRACE] ES ACL update is busy by another goroutine ...")
		time.Sleep(2 * time.Second)
		return resourceElasticsearchACLUpdate(ctx, d, m)
	}
	defer kafkaTopicAvailabilitySem.Release(1)
...
}
```

Reply:
Is that simpler though? I just copied straight from the terraform sdk doc's I.have to admit

Reply:
Internal package seems appropriate because we do not want to export it. But I'm not passionate about it should I move it to pkg/mutexkv?